build/tasks/build.js

```diff
@@ -0,0 +1,29 @@
+var gulp = require('gulp');
+var runSequence = require('run-sequence');
+var paths = require('../paths');
+var tsproject = require( 'tsproject' );
+var debug = require('gulp-debug');
+var ignore = require('gulp-ignore');
+var rename = require('gulp-rename');
+var tsd = require('gulp-tsd');
+
+gulp.task('tsd', function (callback) {
+  console.log('\nRunning tsd\n');
+  return gulp.src('./gulp_tsd.json').pipe(tsd());
+});
+
+gulp.task('compile', ['tsd'], function () {
+  
+  // path to named configuration file provided..
+  return tsproject.src( paths.sourceTsConfig)
+    .pipe(rename({dirname: "",}))
```

Comment:
@richtea Line 19 causes grief with bundle option outDir. Please find a way to overcome this.
With npm release 1.0.5, we will only be using the bundled TsProject. This and a few of the files in the root dir will make up the npm release ( see npmignore )